doc/advanced_configuration.rst

```diff
@@ -520,20 +522,124 @@ The highest precedence is always given to the `-D` flag of the
 ``sphinx-build`` command.
 
 
-.. _find_mayavi:
+.. _image_scrapers:
+
+Image scrapers
+==============
+
+By default, at the end of each code block Sphinx-gallery will detect new
+:mod:`matplotlib.pyplot` figures and turn them into images. This behavior is
+equivalent to the default of::
+
+    sphinx_gallery_conf = {
+        ...
+        'image_scrapers': ('matplotlib',),
+    }
+
+Built-in support is also provided for finding :mod:`Mayavi <mayavi.mlab>`
+figures. To enable this feature, you can do::
+
+    sphinx_gallery_conf = {
+        ...
+        'image_scrapers': ('matplotlib', 'mayavi'),
+    }
+
+.. note:: The parameter ``find_mayavi_figures`` which can also be used to
+          extract Mayavi figures is **deprecated** in version 0.2+,
+          and will be removed in a future release.
+
+Custom scrapers
+^^^^^^^^^^^^^^^
+
+.. note:: The API for custom scrapers is currently experimental.
+
+You can also add your own custom function (or callable class instance)
+to this list. See :func:`sphinx_gallery.gen_rst.matplotlib_scraper` for
+a description of the interface. Here is pseudocode for what this function
+and :func:`sphinx_gallery.gen_rst.mayavi_scraper` do under the hood, which
+uses :func:`sphinx_gallery.gen_rst.figure_rst` to create standardized RST::
+
+    def mod_scraper(block, block_vars, gallery_conf)
+        import mymod
+        image_names = list()
+        image_path_iterator = block_vars['image_path_iterator']
+        for fig, image_path in zip(mymod.get_figures(), image_path_iterator):
+            fig.save_png(image_path)
+            image_names.append(image_path)
+        mymod.close('all')
+        return figure_rst(image_names, gallery_conf['src_dir'])
+
+
+For example, a naive class to scrape any new PNG outputs in the
+current directory could do, e.g.::
+
+    import glob
```

Comment:
Yes let's iterate on the docs, probably after one of the issues listed at the top gets addressed by a user

Reply:
I'm not happy at all with the idea of having a scraper use glob. I think I already raised the concern in another issue. For example, what if people have PNG images on this as data input(scikit-image maybe), then this might accidentally capture those. Or if one day we implement parallel builds, could this lead to problems if some examples save to disk at the same time. But this is an experimental feature, and we will iterate on the docs and inner mechanisms. So I can let this pass.